utils/scripts/visualize_allocations.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+import sys
+from PIL import Image
+
+# This script is used to visualize memory allocations in the Glow compiler.
+#
+# Usage: ./visualize.py dump.txt
+#
+# The script will dump a sequence of bitmap files that can be combined into a
+# video. Example: heap100123.bmp, heap heap100124.bmp, heap100125.bmp ...  )
+#
+# On mac and linux this command will generate a gif file:
+#    convert -delay 10 -loop 0 *.bmp video.gif
+#
+# The input file should contain a list of allocation/deallocation commands.
+# Allocation commands (marked with the letter 'a') report the start address and
+# the size of the buffer, and the deallocation commands (marked with the letter
+# 'd') report the address of the buffer. You can generate these command lists
+# by inserting printf calls into the Glow memory allocator.
+#
+# Example input:
+#    a 348864 20000
+#    a 368896 20000
+#    a 388928 20000
+#    a 408960 200000
+#    d 388928
+#    d 368896
+#    d 348864
```

Comment:
Is there a magic command to produce this input, or we have to instrument the compiler?
In other words, a quick how to generate the input would be nice :).

Reply:
@qcolombet I just added printfs into the memory allocator.

Reply:
I was hoping you knew a magic tool I didn't know :).